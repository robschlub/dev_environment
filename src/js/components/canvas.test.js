import React from 'react';
import renderer from 'react-test-renderer';
import Canvas from './canvas';

describe('Canvas Component', () => {
  test('No props', () => {
    const component = renderer.create(<Canvas />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('With id', () => {
    const component = renderer.create(<Canvas id='test_id' />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('function is called after render', () => {
    const func = jest.fn();
    const component = renderer.create(<Canvas id='test_id' didMountFn={func} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    expect(func.mock.calls).toHaveLength(1);
  });
});
