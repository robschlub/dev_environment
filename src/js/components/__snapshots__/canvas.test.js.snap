// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Canvas Component No props 1`] = `
<div
  id="_container"
>
  <canvas
    className="diagram_gl"
  />
  <div
    className="diagram_html"
  />
  <canvas
    className="diagram_text"
  />
</div>
`;

exports[`Canvas Component With id 1`] = `
<div
  id="test_id_container"
>
  <canvas
    className="diagram_gl"
  />
  <div
    className="diagram_html"
  />
  <canvas
    className="diagram_text"
  />
</div>
`;

exports[`Canvas Component function is called after render 1`] = `
<div
  id="test_id_container"
>
  <canvas
    className="diagram_gl"
  />
  <div
    className="diagram_html"
  />
  <canvas
    className="diagram_text"
  />
</div>
`;
